nexus/sdk-go.md

```diff
@@ -0,0 +1,352 @@
+	"go.temporal.io/sdk/client"
+)
+
+// NewWorkflowRunOperation map an operation to a workflow run.
+func NewWorkflowRunOperation[I, O any](
+	name string,
+	workflow func(internal.Context, I) (O, error),
+	getOptions func(context.Context, I, nexus.StartOperationOptions) (client.StartWorkflowOptions, error),
+) nexus.Operation[I, O]
+
+type WorkflowRunOperationOptions[I, O any] struct {
+	// Operation name.
+	Name string
+	// Workflow function to map this operation to. The operation input maps directly to workflow input.
+	// The workflow name is resolved as it would when using this function in client.ExecuteOperation.
+	// OptionsGetter must be provided when setting this option. Mutually exclusive with Handler.
+	Workflow func(internal.Context, I) (O, error)
+	// Options for starting the workflow. Must be set if Workflow is set. Mutually exclusive with Handler.
+	OptionsGetter func(context.Context, I, nexus.StartOperationOptions) (client.StartWorkflowOptions, error)
+	// Handler for starting a workflow with a different input than the operation. Mutually exclusive with Workflow.
+	Handler func(context.Context, I, nexus.StartOperationOptions) (WorkflowHandle[O], error)
+}
+
+// NewWorkflowRunOperation map an operation to a workflow run with the given options.
+func NewWorkflowRunOperationWithOptions[I, O any](opts WorkflowRunOperationOptions[I, O]) nexus.Operation[I, O]
+
+type WorkflowHandle[T any] interface {
+	ID() string
+	RunID() string
+}
+
+// ExecuteWorkflow starts a workflow run, linking the execution chain to a Nexus operation (subsequent runs started from
+// continue-as-new and retries).
+// Automatically propagates the callback and request ID from the nexus options to the workflow.
+func ExecuteWorkflow[I, O any, WF func(internal.Context, I) (O, error)](
+	ctx context.Context,
+	nexusOptions nexus.StartOperationOptions,
+	startWorkflowOptions client.StartWorkflowOptions,
+	workflow WF,
+	arg I,
+) (WorkflowHandle[O], error)
+
+// ExecuteUntypedWorkflow starts a workflow with by function reference or string name, linking the execution chain to a
+// Nexus operation.
+// Useful for invoking workflows that don't follow the single argument - single return type signature.
+// See [ExecuteWorkflow] for more information.
+func ExecuteUntypedWorkflow[R any](
+	ctx context.Context,
+	nexusOptions nexus.StartOperationOptions,
+	startWorkflowOptions client.StartWorkflowOptions,
+	workflow any,
+	args ...any,
+) (WorkflowHandle[R], error)
+```
+
+**Usage**:
+
+```go
+import (
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporalnexus"
+)
+
+func MyHandlerWorkflow(workflow.Context, MyInput) (MyOutput, error)
+func MyHandlerWorkflowWithAlternativeInput(workflow.Context, MyWorkflowInput) (MyOutput, error)
+
+// Alternative 1 - shortest form, for workflows that have input and outputs that map 1:1 with the operation's I/O.
+opStartTransactionAlt1 := temporalnexus.NewWorkflowRunOperation(
```

Comment:
To clarify, the only reason this exists is for people not wanting to make an options object? But we still make them make a full workflow start options callback? Did they really save much? Maybe instead you should have a `NewWorkflowRunOperationOptions` that accepts these fields instead of top-level overloads.

Reply:
Yes, that's the reason. I believe this will be the preferred way to expose workflows as operations so I made this shorthand form.

The reason I don't like the options struct is that you have to explicitly type the type params but yes a `NewWorkflowRunOperationOptions` also solves that. But keep in mind that even `NewWorkflowRunOperationOptions` may require overloads to support handler and (in the future) a result mapper.

Reply:
> The reason I don't like the options struct is that you have to explicitly type the type params

I don't think this is that bad of a thing in Go and I think we're adding extra methods do to this personal dislike. But I don't have a super strong opinion just so long as we have proper options-struct-based form we can grow on all of these.